### PR TITLE
Filebeat gcs input addFiledJobs panic protection

### PR DESCRIPTION
**Pull request by adassow**
In case `obj, err := s.bucket.Object(name).Attrs(ctx)` return an error, obj will be probably `nil` in consequence the code below will panic 
`objectURI := "gs://" + s.src.BucketName + "/" + obj.Name`

## Proposed commit message
Filebeat gcs input addFiledJobs panic protection

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist
No idea

## How to test this PR locally
No idea

## Related issues
No idea

## Use cases
No idea

## Logs
```
Input crashed with: input panic with: runtime error: invalid memory address or nil pointer dereference
goroutine 938 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/elastic/beats/v7/filebeat/input/v2/input-cursor.(*managedInput).runSource.func1()
	github.com/elastic/beats/v7/filebeat/input/v2/input-cursor/input.go:143 +0x5d
panic({0x558d0f92f8a0, 0x558d123846f0})
	runtime/panic.go:884 +0x213
github.com/elastic/beats/v7/x-pack/filebeat/input/gcs.(*scheduler).addFailedJobs(0xc002e6cb40, {0x558d1007a820, 0xc002e720a0}, {0xc0042de240, 0x3, 0x4})
	github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/scheduler.go:238 +0x378
github.com/elastic/beats/v7/x-pack/filebeat/input/gcs.(*scheduler).scheduleOnce(0xc002e6cb40, {0x558d1007a820?, 0xc002e720a0})
	github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/scheduler.go:106 +0x2d3
github.com/elastic/beats/v7/x-pack/filebeat/input/gcs.(*scheduler).schedule(0xc002e6cb40, {0x558d1007a820?, 0xc002e720a0})
	github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/scheduler.go:60 +0x65
github.com/elastic/beats/v7/x-pack/filebeat/input/gcs.(*gcsInput).Run(_, {0xc006f6a490, {0xc002e70000, 0x86}, {{0x558d0eae5cfe, 0x8}, {0x558d0eae5cfe, 0x8}, {0x558d0eadf24e, 0x6}, ...}, ...}, ...)
	github.com/elastic/beats/v7/x-pack/filebeat/input/gcs/input.go:183 +0x885
github.com/elastic/beats/v7/filebeat/input/v2/input-cursor.(*managedInput).runSource(_, {0xc006f6a490, {0xc002e70000, 0x86}, {{0x558d0eae5cfe, 0x8}, {0x558d0eae5cfe, 0x8}, {0x558d0eadf24e, 0x6}, ...}, ...}, ...)
	github.com/elastic/beats/v7/filebeat/input/v2/input-cursor/input.go:168 +0x479
github.com/elastic/beats/v7/filebeat/input/v2/input-cursor.(*managedInput).Run.func1()
	github.com/elastic/beats/v7/filebeat/input/v2/input-cursor/input.go:122 +0x1bf
github.com/elastic/go-concert/unison.(*MultiErrGroup).Go.func1()
	github.com/elastic/go-concert@v0.2.0/unison/multierrgroup.go:42 +0x75
created by github.com/elastic/go-concert/unison.(*MultiErrGroup).Go
	github.com/elastic/go-concert@v0.2.0/unison/multierrgroup.go:40 +0x8d
```

### CODE REVIEW

**Comment by adassow** [comment]
> This pull request does not have a backport label. If this is a bug or security fix, could you label this PR @adassow? 🙏. For such, you'll need to label your PR with:
> 
> * The upcoming major version of the Elastic Stack
> * The upcoming minor version of the Elastic Stack (if you're not pushing a breaking change)
> 
> To fixup this pull request, you need to add the backport labels for the needed branches, such as:
> 
> * `backport-v8./d.0` is the label to automatically backport to the `8./d` branch. `/d` is the digit

It seems I don't have permission to add any label. It's a bugfix and doesn't change anything except protecting against panic. It shout go probably as a patch.

**Comment by ShourieG** [comment]
Hi @adassow, thanks for the PR, will get it merged asap.

**Review by ShourieG — APPROVED** [review]
LGTM

**Comment by adassow** [comment]
@ShourieG How can I satisfy `buildkite/docs-build-pr` check?

**Comment by ShourieG** [comment]
> @ShourieG How can I satisfy `buildkite/docs-build-pr` check? 

@adassow There seems to be a CI issue on our end, the CI team is looking into it. As soon as it's resolved we'll merge.

**Comment by karenzone** [comment]
buildkite test this rebuild

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -92,6 +92,7 @@ The list below covers the major changes between 7.0.0-rc2 and main only.
 - Make winlogbeat/sys/wineventlog follow the unsafe.Pointer rules. {pull}36650[36650]
 - Cleaned up documentation errors & fixed a minor bug in Filebeat Azure blob storage input. {pull}36714[36714]
 - Fix copy arguments for strict aligned architectures. {pull}36976[36976]
+- Fix filebit gcs input panic {pull}38407[38407]
```
Comment (andrewkroh) [inline CHANGELOG-developer.next.asciidoc:95]: This entry should be added to `CHANGELOG.next.asciidoc`. The "developer" file is focused mainly on API changes or internal-only fixes (without a user impact).

`filebit -> filebeat`

Comment (ShourieG) [reply CHANGELOG-developer.next.asciidoc:95]: I will create a PR to shift this around, since auto merge was enabled and it got merged.

Comment (ShourieG) [reply CHANGELOG-developer.next.asciidoc:95]: https://github.com/elastic/beats/pull/38512

**Pull request merged**